modules/ROOT/pages/anypoint-mq/3.x/anypoint-mq-consume.adoc

```diff
@@ -0,0 +1,92 @@
+endif::[]
+
+The `consume` operation in the Anypoint MQ connector provides the ability to consume a single message at any time of the flow, from any destination, using the acknowledgment strategy that supports your use case.
+
+== Incoming Message Structure
+
+Every consumed message consists of two parts:
+
+* `payload` - Represents the value of the received message body
+* `attributes` - Contains metadata regarding the received message, such as the message properties, or its ID.
+
+image::anypoint-mq/3.x/amq-3x-consume-attributes.png[Anypoint MQ Message Attributes]
+
+
+== Consume a Message
+
+The simplest declaration to consume a message from a queue is to drag the *Consume* operation from the Mule palette to the canvas and configure the `Queue` parameter with the name of the queue from which to retrieve the message:
+
+image::anypoint-mq/3.x/amq-3x-consume-operation.png[Anypoint MQ Consume Operation]
+
+For example:
+
+[source,xml,linenums]
+----
+<anypoint-mq:consume config-ref="Anypoint_MQ_config" destination="myQueue"/>
+----
+
+The Consume operation in this example consumes the first available message in `myQueue` and converts it to a `MuleMessage` with this structure:
+
+* The message's body as payload.
+* The message's metadata in the message attributes. Attributes might include the `messageId` or the `redeliveryCount`.
+
+TIP: Use a DataWeave expression in the `Queue` parameter to dynamically change the target queue from which to consume the message.
+
+
+== Consumed Message Acknowledgment
+
+* *Automatic*
++
+By default, the `acknowledgementMode` used is `IMMEDIATE`, meaning that the message is acknowledged as soon as it is consumed from the queue and before it's delivered to the Mule flow. If the message acknowledgment fails, the `Consume` operation terminates with an `ANYPOINT-MQ:ACKING` error.
++
+For more information, see xref:anypoint-mq/3.x/anypoint-mq-ack.adoc#immediate-acknowledgment[Immediate Acknowledgment].
+
+* *Manual*
++
+To manually control the acknowledgment of the message, change the `acknowledgementMode` to `MANUAL`. When using the `MANUAL` acknowledgment mode, the app logic decides when to perform the acknowledgment of the message, using the xref:anypoint-mq/3.x/anypoint-mq-ack.adoc[ACK] or xref:anypoint-mq/3.x/anypoint-mq-ack.adoc[NACK] operations.
++
+To perform the manual acknowledgment, save the value of the `ackToken` provided as part of the resulting message attributes.
++
+For more information, see xref:anypoint-mq/3.x/anypoint-mq-ack.adoc#manual-acknowledgment[Manual Acknowledgment].
+
+For information on acknowledgment timeouts, see xref:anypoint-mq/3.x/anypoint-mq-ack.adoc#acknowledgment-timeout[Acknowledgment Timeout].
+
+== Manage Empty Queues
+
+When consuming a message from a queue, either the queue has messages, or it is empty.
+
+The first scenario is the most common: a queue has messages waiting to be consumed. When the `Consume` operation executes, it returns immediately with one of the messages.
```

Comment:
I suggest using a bullet list rather than subsections:

"When a queue has messages, the 'Consume' operation immediately delivers a message. When a queue is empty, the operation results immediately with a 'null' message or it waits for a new message to arrive, depending on how you set the 'pollingTime' parameter:

     o 0
        If you set 'pollingTime to 0, the....

     o 1 to 20
        If you set...."